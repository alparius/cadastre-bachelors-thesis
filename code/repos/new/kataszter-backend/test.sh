#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
