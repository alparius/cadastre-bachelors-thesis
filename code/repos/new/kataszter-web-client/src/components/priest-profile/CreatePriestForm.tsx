import React, { useState } from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { Button, Form, Message } from "semantic-ui-react";

import IDate from "../../data/Date";
import ICreatePriestData, { NewCreatePriestData } from "../../data/priest/CreatePriestData";
import ICreatePriestResponse from "../../requests/response/CreatePriestResponse";
import BoldFM from "../../util/styledMessages/BoldFM";
import DateInput from "../DateInput";

interface ICreatePriestFormProps {
    priest: ICreatePriestData;
    response: ICreatePriestResponse | undefined;
    setPriest(priest: ICreatePriestData): void;
    asyncPost(): Promise<void>;
}

const CreatePriestForm: React.FC<ICreatePriestFormProps> = (props: ICreatePriestFormProps) => {
    const { priest, setPriest, response, asyncPost } = props;
    const { formatMessage } = useIntl();

    const [errorMsg, setErrorMsg] = useState(false);

    const handleChange = (event: any) => {
        const { name, value } = event.target;
        setPriest({ ...priest, [name]: value });
    };

    const dateChange = (newDate: IDate) => {
        setPriest({ ...priest, birthDate: newDate });
    };

    const handleSubmit = () => {
        if (priest.name === "" || priest.birthTown === "" || priest.birthDate.year) {
            setErrorMsg(true);
        } else {
            setErrorMsg(false);
            asyncPost();
        }
    };

    return (
        <Form onSubmit={handleSubmit}>
            {response && (
                <Message negative>
                    <FM id={response.label} />
                </Message>
            )}
            {errorMsg && (
                <Message negative>
                    <FM id="formError.invalidOrEmpty" />
                </Message>
            )}
            <Form.Input type="text" name="name" label={formatMessage({ id: "profile.name" }) + " *"} onChange={handleChange} />
            <Form.Field>
                <BoldFM id="profile.birthDate" />
                <b> *</b>
                <DateInput date={priest.birthDate} changeDate={dateChange} />
            </Form.Field>
            <Form.Input type="text" name="birthTown" label={formatMessage({ id: "profile.birthTown" }) + " *"} onChange={handleChange} />
            <Form.Input type="text" name="birthCounty" label={formatMessage({ id: "profile.birthCounty" })} onChange={handleChange} />
            <Form.Input type="text" name="diocese" label={formatMessage({ id: "profile.diocese" })} onChange={handleChange} />
            <Form.Input type="text" name="birthCountry" label={formatMessage({ id: "profile.birthCountry" })} onChange={handleChange} />

            <Button type="submit" className="ui violet button">
                <FM id="insert.priest" />
            </Button>
            <Button type="reset" className="ui button" onClick={() => setPriest(NewCreatePriestData)}>
                <FM id="general.reset" />
            </Button>
        </Form>
    );
};

export default CreatePriestForm;
