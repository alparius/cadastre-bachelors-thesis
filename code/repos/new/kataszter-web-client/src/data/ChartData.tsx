interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
