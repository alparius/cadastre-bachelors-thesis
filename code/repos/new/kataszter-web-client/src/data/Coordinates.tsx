interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
