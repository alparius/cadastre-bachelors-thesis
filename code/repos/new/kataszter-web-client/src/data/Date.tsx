interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
