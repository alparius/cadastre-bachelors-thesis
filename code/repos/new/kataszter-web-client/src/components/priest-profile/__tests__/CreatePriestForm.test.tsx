/*===================
  CreatePriestForm.unit.test.tsx
 ===================*/
import React from "react";

import ICreatePriestData from "../../../data/priest/CreatePriestData";
import ILabelResponse from "../../../requests/response/LabelResponse";
import { NewTestCreatePriestData, NewTestCreateResponse, NewTestLabelResponseError } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntlAndRouter } from "../../../util/test/test-wrapper";
import CreatePriestForm from "../CreatePriestForm";

const handlers = {
    setPriest: jest.fn(),
    asyncPost: jest.fn()
};

const priest: ICreatePriestData = NewTestCreatePriestData;

describe("CreatePriestForm element", () => {
    afterEach(() => {
        handlers.setPriest.mockClear();
    });

    it("should render a form", () => {
        const container = createComponentWithIntlAndRouter(<CreatePriestForm priest={priest} response={undefined} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a message too when there is a response", () => {
        const container = createComponentWithIntlAndRouter(
            <CreatePriestForm priest={priest} response={NewTestCreateResponse} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the setPriest function when the form gets input", () => {
        const container = mountWithIntlAndRouter(<CreatePriestForm priest={priest} response={undefined} {...handlers} />);
        container.find('input[name="name"]').simulate("change", { target: { value: "test" } });
        container.find('input[name="birthCountry"]').simulate("change", { target: { value: "test" } });
        container.find('input[name="birthCounty"]').simulate("change", { target: { value: "test" } });
        container.find('input[name="birthTown"]').simulate("change", { target: { value: "test" } });
        container.find('input[name="diocese"]').simulate("change", { target: { value: "test" } });
        expect(handlers.setPriest).toHaveBeenCalledTimes(5);
    });
});
