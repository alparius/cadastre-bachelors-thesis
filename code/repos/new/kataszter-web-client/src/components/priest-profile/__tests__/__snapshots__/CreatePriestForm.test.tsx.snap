// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePriestForm element should render a form 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label>
      Full name *
    </label>
    <div
      className="ui input"
    >
      <input
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <b>
      Date of birth
    </b>
    <b>
       *
    </b>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="inline unstackable fields"
        style={
          Object {
            "marginBottom": "-2px",
          }
        }
      >
        <div
          className="field"
        >
          <b
            style={Object {}}
          >
            year
          </b>
          <div
            className="ui fluid input"
            style={
              Object {
                "width": "6em",
              }
            }
          >
            <input
              id=""
              name="year"
              onChange={[Function]}
              type="number"
              value={2000}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <b
            style={Object {}}
          >
            month
          </b>
          <div
            className="ui fluid input"
            style={
              Object {
                "width": "5em",
              }
            }
          >
            <input
              id=""
              name="month"
              onChange={[Function]}
              type="number"
              value={6}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <b
            style={Object {}}
          >
            day
          </b>
          <div
            className="ui fluid input"
            style={
              Object {
                "width": "5em",
              }
            }
          >
            <input
              id=""
              name="day"
              onChange={[Function]}
              type="number"
              value={6}
            />
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <i
          style={
            Object {
              "color": "gray",
            }
          }
        >
          comment about date (optional)
        </i>
        <div
          className="ui fluid input"
        >
          <input
            id="grey"
            name="comment"
            onChange={[Function]}
            type="text"
            value="2000.06.06"
          />
        </div>
      </div>
    </form>
  </div>
  <div
    className="field"
  >
    <label>
      Town of birth *
    </label>
    <div
      className="ui input"
    >
      <input
        name="birthTown"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      County of birth
    </label>
    <div
      className="ui input"
    >
      <input
        name="birthCounty"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Diocese
    </label>
    <div
      className="ui input"
    >
      <input
        name="diocese"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Country of birth
    </label>
    <div
      className="ui input"
    >
      <input
        name="birthCountry"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <button
    className="ui button ui violet button"
    onClick={[Function]}
    type="submit"
  >
    Add new priest
  </button>
  <button
    className="ui button ui button"
    onClick={[Function]}
    type="reset"
  >
    Reset
  </button>
</form>
`;

exports[`CreatePriestForm element should render a message too when there is a response 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui negative message"
  >
    info.test
  </div>
  <div
    className="field"
  >
    <label>
      Full name *
    </label>
    <div
      className="ui input"
    >
      <input
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <b>
      Date of birth
    </b>
    <b>
       *
    </b>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="inline unstackable fields"
        style={
          Object {
            "marginBottom": "-2px",
          }
        }
      >
        <div
          className="field"
        >
          <b
            style={Object {}}
          >
            year
          </b>
          <div
            className="ui fluid input"
            style={
              Object {
                "width": "6em",
              }
            }
          >
            <input
              id=""
              name="year"
              onChange={[Function]}
              type="number"
              value={2000}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <b
            style={Object {}}
          >
            month
          </b>
          <div
            className="ui fluid input"
            style={
              Object {
                "width": "5em",
              }
            }
          >
            <input
              id=""
              name="month"
              onChange={[Function]}
              type="number"
              value={6}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <b
            style={Object {}}
          >
            day
          </b>
          <div
            className="ui fluid input"
            style={
              Object {
                "width": "5em",
              }
            }
          >
            <input
              id=""
              name="day"
              onChange={[Function]}
              type="number"
              value={6}
            />
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <i
          style={
            Object {
              "color": "gray",
            }
          }
        >
          comment about date (optional)
        </i>
        <div
          className="ui fluid input"
        >
          <input
            id="grey"
            name="comment"
            onChange={[Function]}
            type="text"
            value="2000.06.06"
          />
        </div>
      </div>
    </form>
  </div>
  <div
    className="field"
  >
    <label>
      Town of birth *
    </label>
    <div
      className="ui input"
    >
      <input
        name="birthTown"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      County of birth
    </label>
    <div
      className="ui input"
    >
      <input
        name="birthCounty"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Diocese
    </label>
    <div
      className="ui input"
    >
      <input
        name="diocese"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Country of birth
    </label>
    <div
      className="ui input"
    >
      <input
        name="birthCountry"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <button
    className="ui button ui violet button"
    onClick={[Function]}
    type="submit"
  >
    Add new priest
  </button>
  <button
    className="ui button ui button"
    onClick={[Function]}
    type="reset"
  >
    Reset
  </button>
</form>
`;
